packages/caver-core-method/src/index.js

```diff
@@ -764,26 +764,35 @@ const checkForContractDeployment = (mutableConfirmationPack, receipt, sub) => {
 
     if (!receipt.status && receipt.txError) {
```

Comment:
```suggestion
    if (!Boolean(utils.hexToNumber(receipt.status)) && receipt.txError) {
```

Reply:
The `receipt.status` already has a value of boolean type and was not touched in this PR. Do you want to change as you suggested?

Reply:
When i test this, `receipt.status` was in hex string, so even transaction was failed though, still did not go into this logic.

Reply:
The output of `typeof receipt.status` was a boolean type. Am I wrong? If so, could you let me know how did you check that type?

> so even transaction was failed though, still did not go into this logic.

Seems strange. I saw the same output as the `master`' branch and this PR.

Reply:
I guess you are using before common architecture feature?
Depends on what you are using, `receipt.status` field can be boolean type (this is depends on outputFormatter of the rpc call).
```
    const caver = new Caver('http://localhost:8551')
    const keyring = caver.wallet.add(
        caver.wallet.keyring.createFromPrivateKey('0x4367aeca3984c23f4d6b1b35a0edaa195d7847536f99a50a0bf178330a0ce075')
    )
    const tx = caver.transaction.smartContractDeploy.create({
        from: keyring.address,
        gas: 3000000,
        value: 0,
        input: '0x...'
    })
    await caver.wallet.sign(keyring.address, tx)
    console.log(await caver.rpc.klay.sendRawTransaction(tx))
```

And if recipet.status is boolean type though, suggested logic still works, so i think not big deal.

Reply:
Yes, I used the `deploy()` function and I didn't know there were multiple ways to do that. I was not against your suggestion but wanted to know what we saw as a difference.